Library/Homebrew/keg_relocate.rb

```diff
@@ -22,7 +24,7 @@ def freeze
 
     sig { params(key: Symbol, old_value: T.any(String, Regexp), new_value: String).void }
     def add_replacement_pair(key, old_value, new_value)
-      @replacement_map[key] = [old_value, new_value]
+      @replacement_map[key] = [self.class.path_regex(old_value), new_value]
```

Comment:
Does this affect the `start_with?` tests in [`relocate_dynamic_linkage`](https://github.com/Homebrew/brew/blob/2b9fd5460e357b6638944f4f89917a669a5af68c/Library/Homebrew/extend/os/mac/keg_relocate.rb#L21)?

Reply:
Ugh, yeah it does.

Also, do we still want this behavior when `old_value` is a placeholder in [`prepare_relocation_to_locations`](https://github.com/Homebrew/brew/blob/2b9fd5460e357b6638944f4f89917a669a5af68c/Library/Homebrew/keg_relocate.rb#L94-L102)? I don't think it will cause any problems because the only places a placeholder would be is where the path is already guaranteed to match the regex, but it feels a bit weird.

Reply:
Actually, I think we _don't_ want this regex for pathnames because it won't properly replace the placeholders for any existing bottles that have the placeholders in the middle (e.g. `objc-run`).